### PR TITLE
sci-physics/rivet: new package, add 3.1.5, 3.1.6

### PR DESCRIPTION
**Pull request by APN-Pucky**
I keep this as a draft until the dependencies 

- [x] https://github.com/gentoo/gentoo/pull/26613
- [x] https://github.com/gentoo/gentoo/pull/26618

are merged.

Signed-off-by: Alexander Puck Neuwirth <alexander@neuwirth-informatik.de>

### CODE REVIEW

File: sci-physics/rivet/rivet-3.1.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+REQUIRED_USE="
+	^^ ( hepmc3 hepmc2 )
+	${PYTHON_REQUIRED_USE}
+"
+
+RDEPEND="
+	>=sci-physics/yoda-1.9.4[python(-),${PYTHON_SINGLE_USEDEP}]
+	>=sci-physics/fastjet-3.4.0[plugins]
+	>=sci-physics/fastjet-contrib-1.046
+	hepmc2? ( sci-physics/hepmc:0=[-cm(-),gev(+)] )
```
Comment (AndrewAmmerlaan) [inline sci-physics/rivet/rivet-3.1.5.ebuild:31]:
```suggestion
	hepmc2? ( sci-physics/hepmc:2=[-cm(-),gev(+)] )
```

File: sci-physics/rivet/rivet-3.1.6.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+REQUIRED_USE="
+	^^ ( hepmc3 hepmc2 )
+	${PYTHON_REQUIRED_USE}
+"
+
+RDEPEND="
+	>=sci-physics/yoda-1.9.5[python(-),${PYTHON_SINGLE_USEDEP}]
+	>=sci-physics/fastjet-3.4.0[plugins]
+	>=sci-physics/fastjet-contrib-1.048
+	hepmc2? ( sci-physics/hepmc:0=[-cm(-),gev(+)] )
```
Comment (AndrewAmmerlaan) [inline sci-physics/rivet/rivet-3.1.6.ebuild:31]:
```suggestion
	hepmc2? ( sci-physics/hepmc:2=[-cm(-),gev(+)] )
```

Comment (APN-Pucky) [reply sci-physics/rivet/rivet-3.1.6.ebuild]: Thanks for spotting that. I wrote it before the slot 2 existed.
I have an similar ebuild for pythia with `hepmc2` and `hepmc3` flags. Should I PR that also to `::gentoo`? Though it invalidates/replaces the existing `hepmc` flag.

Comment (AndrewAmmerlaan) [reply sci-physics/rivet/rivet-3.1.6.ebuild]: > Should I PR that also to `::gentoo`? Though it invalidates/replaces the existing `hepmc` flag.

Sure! :+1:

**Comment by AndrewAmmerlaan** [comment]
Nice work, Thanks :+1:

**Pull request closed without merging**